File: smallest_window_containing_012.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
// User function Template for C++

class Solution {
  public:
    int smallestSubstring(string S) {
        // Code here
        int one=-1,zero=-1,two=-1,ans=INT_MAX;
        for(int i=0;i<S.size();i++){
            if(S[i]=='0') zero=i;
            else if(S[i]=='1') one=i;
            else two=i;
            if(one!=-1 && two !=-1 && zero!=-1){
                int mn=min({one,two,zero});
                int mx=max({one,two,zero});
                ans=min(ans,(mx-mn+1));
            }
        }
        return ans==INT_MAX?-1:ans;
    }
};


//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        string S;
        cin >> S;
        Solution ob;
        cout << ob.smallestSubstring(S);
        cout << endl;
    }
}
// } Driver Code Ends

File: smallest_window_containing_012.py
#User function Template for python3

class Solution:
    def smallestSubstring(self, S):
        one, zero, two = -1, -1, -1
        ans = float('inf')

        for i in range(len(S)):
            if S[i] == '0':
                zero = i
            elif S[i] == '1':
                one = i
            else:
                two = i
            
            if one != -1 and two != -1 and zero != -1:
                mn = min(one, two, zero)
                mx = max(one, two, zero)
                ans = min(ans, (mx - mn + 1))
        
        return -1 if ans == float('inf') else ans


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
	t=int(input())
	for i in range(t):
		S = input()
		ob = Solution()
		ans = ob.smallestSubstring(S)
		
		print(ans)



# } Driver Code Ends